core/src/main/java/hudson/util/jna/GNUCLibrary.java

```diff
@@ -89,11 +90,17 @@
     // this is listed in http://developer.apple.com/DOCUMENTATION/Darwin/Reference/ManPages/man3/sysctlbyname.3.html
     // but not in http://www.gnu.org/software/libc/manual/html_node/System-Parameters.html#index-sysctl-3493
     // perhaps it is only supported on BSD?
+    @Deprecated
     int sysctlbyname(String name, Pointer oldp, IntByReference oldlenp, Pointer newp, IntByReference newlen);
+    int sysctlbyname(String name, Pointer oldp, NativeLongByReference oldlenp, Pointer newp, NativeLong newlen);
 
+    @Deprecated
     int sysctl(int[] mib, int nameLen, Pointer oldp, IntByReference oldlenp, Pointer newp, IntByReference newlen);
+    int sysctl(int[] name, int namelen, Pointer oldp, NativeLongByReference oldlenp, Pointer newp, NativeLong newlen);
 
+    @Deprecated
     int sysctlnametomib(String name, Pointer mibp, IntByReference size);
+    int sysctlnametomib(String name, Pointer mibp, NativeLongByReference sizep);
```

Comment:
@ngg Could you please add some comments explaining that the use of `NativeLongByReference` and `NativeLong` relies on an implementation detail that is always true in practice and should be replaced with `size_t` and `size_t.ByReference` when we upgrade to JNA 5.9.0 and add `jna-platform` to the build?

Reply:
I've added a comment about this.